### PR TITLE
vote notification

### PR DESCRIPTION
**Pull request by jcvernaleo**
This PR is still WIP

It goes with decred/dcrrpcclient#29 to address decred/haste#1

### CODE REVIEW

File: rpcwebsocket.go [hunk 1]
```diff
@@ -228,6 +230,20 @@ func (m *wsNotificationManager) NotifyBlockDisconnected(block *dcrutil.Block) {
 	}
 }
 
+// NotifyVotes passes a block newly-connected to the best chain
+// to the notification manager for block and transaction notification
+// processing.
+func (m *wsNotificationManager) NotifyBlockConnected(block *dcrutil.Block) {
```
Comment (chappjc) [inline rpcwebsocket.go]: copy/paste error?  or not yet implemented?

    NotifyBlockConnected vs. NotifyVotes

Comment (jcvernaleo) [inline rpcwebsocket.go]: Kind of a little of both.  The PR isn't done and that was a copy/paste error.

**Comment by raedah** [comment]
What will this be used for?

**Comment by chappjc** [comment]
@raedah For one, the next version is dcrspy.  I'll have this be an email notification event for sure.

But what actions would the core software take?  Dunno.

**Comment by jcvernaleo** [comment]
I mainly want it for mining related things but I believe we've talked about adding it for a while since it is something that could be used in other things.

**Comment by cjepson** [comment]
It's useful in telling a PoW miner that they need to call getwork again, to get a new block template that has more votes.

**Comment by chappjc** [comment]
Will it notify if the vote originates locally (dcrwallet RPC)?

File: dcrjson/chainsvrwsntfns.go [hunk 2]
```diff
@@ -109,6 +113,23 @@ func NewReorganizationNtfn(oldHash string, oldHeight int32, newHash string,
 	}
 }
 
+// NewVoteNtfn defines the newvote JSON-RPC notification.
+type NewVoteNtfn struct {
+	VoteHash  string
+	BlockHash string
+	Vote      bool
+}
```
Comment (chappjc) [inline dcrjson/chainsvrwsntfns.go]: ticket hash?
anything else from the ssgen like subsidy address?

**Comment by jcvernaleo** [comment]
I think (although I haven't really confirmed) that what I was working on would only care that the daemon saw a vote, not where it came from.
Probably will include ticket hash.  Wasn't planning on subsidy address.  Also probably best to ask these things when the PR is actually ready for review and not still in this only partially done state.

**Comment by chappjc** [comment]
@jcvernaleo OK.  I was actually thinking it was better to ask now while the PR is still being implemented, before the code becomes potentially hard to change.  But I get what you mean.  I'll check back later.  :)

**Comment by jcvernaleo** [comment]
@chappjc Definitely understand that.  And to a certain degree that is why I pushed the PR while it was completely unfinished.  But I didn't want to give the impression that this was nearly done (there is another task ahead of it in my todo list at the moment).

**Comment by jcvernaleo** [comment]
Just wanted to point out here that this PR will be worked on again soon.  I initially opened it way earlier than normal so I could get feedback (which I got and will be useful when I finish it up).

**Comment by jcvernaleo** [comment]
This will (when we get back to it) close #455

**Comment by jcvernaleo** [comment]
Closing this since it is on hold.  Can always reopen when/if we get back to it.

**Pull request closed without merging**